File: be-api/src/main/java/cd/beapi/dto/response/StaffResponse.java
package cd.beapi.dto.response;

import cd.beapi.enumerate.StaffType;

import java.time.Instant;
import java.time.LocalDate;

public record StaffResponse(
        Long id,
        String code,
        String fullName,
        String phone,
        String email,
        LocalDate dob,
        Boolean gender,
        String address,
        String avatarUrl,
        StaffType staffType,
        Boolean isActive,
        Long version,
        Instant createdAt,
        Instant modifiedAt,
        Long userId,
        String username,
        String temporaryPassword) {
}

File: fe-admin/src/app/core/model/response/staff-response.ts
export interface StaffResponse {
    id: number;
    code: string;
    fullName: string;
    phone: string;
    email: string;
    dob: Date;
    gender: boolean | null;
    address: string;
    avatarUrl: string | null;
    staffType: string;
    isActive: boolean;
    version: number;
    createdAt: Date;
    modifiedAt: Date;
}
